nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java

```diff
@@ -129,9 +129,10 @@ public Description createErrorDescription(
       checkName = INITIALIZATION_CHECK_NAME;
     }
 
-    // Mildly expensive state.getPath() traversal, occurs only once per potentially
-    // reported error.
-    if (hasPathSuppression(state.getPath(), checkName)) {
+    // Mildly expensive state.getPath() traversal, occurs twice per potentially reported error.  We
+    // can get rid of the check for "all" once we require Error Prone 2.22.0 or higher.
```

Comment:
This is what I was talking about on the EP support policy, we probably want to remove this once we drop support for older EP versions (and we can have an issue to track it due to performance implications)

Reply:
This is a very good question.  Right now, we are stuck supporting a very old EP version (2.10.0) since we still support running NullAway on JDK 8, and that was the last EP version that supported running on JDK 8.  At this point, I think we should make a plan for dropping JDK 8 support in the near-ish future, and then we could bump our minimum EP version to something much more recent.  As to what policy we should adhere to after that, I'm thinking something like we will typically support an EP version for X months after it was released.  For reference, 2.10.0 was [released](https://github.com/google/error-prone/releases/tag/v2.10.0) on Nov. 4, 2021, so we are just past two years since it came out.  Maybe we could generally aim for supporting an EP release for 12-18 months?  Not sure of the right number, and we don't need a hard-and-fast rule.  But I agree we should have something in mind.

I've added a comment on https://github.com/uber/NullAway/issues/634 regarding dropping JDK 8 support.  And I've opened https://github.com/uber/NullAway/issues/882 regarding support for older Error Prone versions).